### DISCUSSION TITLE
Dynamical Systems & Theoretical Aspects of Life

### DISCUSSION THREAD

**Discussion by BrainAnnex** [BrainAnnex/life123#20] [Ideas]
**Diffusion** by itself is a dynamical system that tends to equilibrium and is governed by partial differential equations.
**Chemical reactions**, on a similar note, are a dynamical system that the is governed by ordinary differential equations, and also moves towards an equilibrium...  (Somewhat simplifying here, but bear with me.)

And that would be the end of the story for each of them, if there were by themselves!

But the moment that **diffusion and reactions come together** then we have 2 linked dynamical systems: each moving towards its own equilibrium - but, in the process of doing so, affecting and perturbing the path to equilibrium of the other dynamical system!   This constant interference is what leads to a richness of behavior that we're interested in exploring.

### Compartments

And then the next layer of "magic" happens: when **compartments** (membranes) arise.

And, fascinatingly, they arise naturally, thanks to quirks of chemistry, such as a carbons forming polymers easily, and water affecting the behavior of polar groups in molecules.  And once we have compartments - as in biological cells and subcellular compartments, then the real magic of biology can start...

How so?  Well, the formation of compartments, in turn, profoundly affects the dynamical system of diffusion, because now diffusion rates will change.  And they will now differ in different locations.  In particular, diffusion across membrane boundaries, will typically be different, or it may not even happen at all, or may only happen for particular chemical species, or combinations of species when they react or band together as a complex that can cross the membrane.  

And at this point, this triple system (**diffusion + reactions + compartments**) presumably holds the foundational ingredients to create some interesting "pre-biology".

I'd like to see a deeper exploration of this dynamical system, including **attractors and chaotic modalities**, as well as more investigation of "dynamical systems morphing into circuits (blocks of functionality of interest)"

And I'd love to see a **Machine Learning** approach, defining a Gain function based on some suitably-defined "interesting" behavior of the system, such as a time evolution that is "reminiscent" of real biological systems.  In particular, the compartments themselves (membranes) growing, shrinking, pinching off or fusing.

### From Analog to Analog+Digital

Interestingly, everything discussed so far is "analog".  But the plot thickens the moment that the simulations start dealing with "guided polymerization reactions" - i.e. macromolecules replicating and guiding the synthesis of other macromolecules (as DNA does with proteins): at that point, a "digital" element emerges.

A dynamical system like what we have conceptually built so far, presumably has the capability of generating a variety - possibly a very wide variety - of behaviors.  At which point does a dynamical system cross is over into the _emergent property_ of being able to be regarded as a _"computing device"_ ?

In Life123, our starting point is diffusion and reactions, to which we will be gradually adding other elements (starting with membranes and compartmentalization, later followed by macromolecules that replicate and guide the formation of other macromolecules )...  At which point along the series of _more biological-like_ simulations, do we cross that threshold into realizing some kind of computing device that can be regarded as having a "purpose" that we tend to identify with life?

### From Dynamical Systems to Computing devices
Intuitively, the ability to carry on some kind of computation is to us the hallmark of what could be described is having a "purpose" that transcends being a mere, albeit possibly complex, dynamical system : intuitively, doing something, accomplishing something, not just "being" or "going with the flow."

According to a fascinating book by someone who has long thought about all this, [The Nature of Physical Computation, by Oron Shagrir, Oxford University Press (2022)](https://www.amazon.com/Physical-Computation-Studies-Philosophy-Science-ebook/dp/B09LVV662Y), a theoretical definition of computation may best be reserved for a physical system that, in addition to implementing some kind of a function (such as the state evolution of a dynamical system), also performs some kind of **modeling**.  Modeling, loosely speaking, is the carrying out of a computation that mimics salient features of _another_ system.  

Shagrir, the author of the above book, gives extensive justification for his generalized view of what is most sensibly regarded a computing device, above and beyond the more limiting confines of **Turing machines** and similar conceptual devices.

The "ability to model another system" is indeed the bread-and-butter of biological cells.  After all, they can be quite adept at assessing the environment - and taking remedial steps based on the past circumstances - steps that are most conducive to the system retaining some kind of identity and homeostasis.  **Essentially, biological cells may be seen as dynamical systems that can model their environment.**  That would seem to fit in well with Shagrir's definition of a computing device, as I understand it.

That ability to model (predict/anticipate/react to) their environment is what give microorganisms what we may anthropomorphize as "having a purpose", and "trying to stay alive".

The grand variety of possible "computing systems" - broadly defined - in biological cells and in our simulated ones, is the raw material for evolution (Machine Learning) to traverse a search space and lead to particularly interesting complex systems - especially systems that closely approximate real-life prokaryotic, and later eukaryotic, cells - and that's what we are keen to explore!

**Comment by mihaitodor** [comment]
Very cool! This introduction reminded me of a few of Sean Carroll's Mindscape [podcasts](https://www.preposterousuniverse.com/podcast/) that I listened to recently:
- Stuart Bartlett on What “Life” Means: https://www.preposterousuniverse.com/podcast/2020/07/20/106-stuart-bartlett-on-what-life-means/ He tries to define a framework for what would be the minimum requirements for a system to be considered "life".
- Nick Lane on Powering Biology: https://www.preposterousuniverse.com/podcast/2022/05/23/198-nick-lane-on-powering-biology/ He explains the Krebs cycle and reverse Krebs Cycle and how cellular membranes work. I need to listen to this one again and read some of his work.

Sean had several other guests talking about biology, chemistry and biophysics, but I think the above two are the most relevant here.

**Comment by BrainAnnex** [comment]
I have further explored this foundational, monumental topic with ChatGPT.

Here's an extract from the start of our chat thread:

> Let's talk about "emergent phenomena" : humans have long marveled at particle physics giving rise to chemistry, at chemistry giving rise to biology, at biology giving rise to intelligence. Plus the emergence of weather, statistical mechanics, etc. All natural instances of emergence. 
> But now we also have the human-made emergent phenomena of text-predicting RNN's - basically "statistical parrots" giving rise to very sophisticated LLM, such as yourself, that can make complex and/or creative inferences. As we marvel at both natural and artificial emergent phenomena, I get the impression that we still profoundly lack any good theory.

See the full, breath-taking and very insightful, [chat here](https://chatgpt.com/share/69db3ece-38e4-83e8-84f8-90d0697b1bfc)!
